repository-hbase/src/main/java/hbase/load/BulkLoader.java

```diff
@@ -89,15 +88,17 @@ public int run(String[] strings) throws Exception {
             System.exit(ERROR);
         }
 
+        getConf().set(REVERSE_FLAG, strings[ARG_REVERSE_FLAG]);
+
         // Populate map reduce
         getConf().set(ROWKEY_POSITION, strings[ARG_CSV_ROWKEY_POSITION]);
-//         String headerArg = strings[ARG_CSV_HEADER_STRING];
-//         if (headerArg.isEmpty()) {
-               Configuration conf = this.getConf();
-               getHeader(strings[ARG_CSV_FILE], conf);
-//         } else {
-//               getConf().set(HEADER_STRING, strings[ARG_CSV_HEADER_STRING]);
-//         }
+
+        if (strings.length == MAX_ARGS) {
```

Comment:
This needs to change as it looks like we now have two optional args i.e. the header and the hfile directory (if no hfile directory then we insert straight into hbase). That is probably why the tests are failing because for those we load direct in Hbase and not via HFiles. Hvae a think how we can support multiple optinal args. Maybe if we pass "useFileColumnHeadings" or the actual headings then the headers becomes mandatory and only the hfile directory is optional